src/DurableTask.SqlServer/SqlUtils.cs

```diff
@@ -398,7 +398,7 @@ static IEnumerable<SqlDataRecord> GetInstanceIdRecords(IEnumerable<string> insta
                 command,
                 traceHelper,
                 instanceId,
-                cmd => cmd.ExecuteReaderAsync(cancellationToken));
+                cmd => cmd.ExecuteReader());
```

Comment:
Instead of making this change for all calls to `ExecuteReaderAsync`, I would rather we make a separate `ExecuteReader` API and selectively choose which operations use sync vs. async. My preference is that `IOrchestrationService` interface methods use the synchronous variant whereas the `IOrchestrationServiceClient` methods continue to use the async variant. The reason for the latter is that an unknown number of threads may try to call `IOrchestrationServiceClient` methods, and I want to avoid the possibility of running into thread starvation for those calls.